sounddevice.py

```diff
@@ -2349,6 +2376,97 @@ def __init__(self, channel_selectors):
+            device.  *channel_map* is a list of integers specifying the
+            (zero-based) channel numbers to use.  Note that
+            *channel_map* is treated differently between input and
+            output channels.  See the example below.
+        change_dev_params : bool, optional
+            If ``True``, allows PortAudio to change things like the
+            device's frame size, which allows for much lower latency,
+            but might disrupt the device if other programs are using it,
+            even when you are just Querying the device.  ``False`` is
+            the default.
+        prevent_rate_conv : bool, optional
+            In combination with the above flag, ``True`` causes the
+            stream opening to fail, unless the exact sample rates are
+            supported by the device.
+        rate_conv_quality : string, optional
+            This sets MacCore's sample rate conversion quality.
+            Possible values are: ``'Min'``, ``'Low'``, ``'Medium'``,
+            ``'High'``, or ``'Max'``.  ``'Max'`` is the default.
+
+        Example
+        -------
+        Input from 'Mic 9' and 'Mic 10' on RME Fireface UFX+, and send
+        output to 'Analog 3' and 'Analog 4':
+
+        >>> import sounddevice as sd
+        >>>
+        >>> input_channels = [8, 9]
+        >>> output_channels = [2, 3]
+        >>>
+        >>> # For input devices, the channel_map is simply the list of
+        >>> # 0-based index numbers for each channel:
+        >>> ch_map_in = input_channels
+        >>> mc_in = sd.MacCoreSettings(channel_map=ch_map_in)
+        >>>
+        >>> # For output devices, the list needs to be "inverted":
+        >>> # Populate the array with -1 for unwanted channels:
+        >>> max_output_channels = query_devices(device, 'output')['max_output_channels']
+        >>> ch_map_out = [-1 for _ in range(max_output_channels)]
+        >>> # For this example device, max_output_channels == 94.
+        >>> # Now overwrite any -1's for channels we'll use for output:
+        >>> for ndx in range(len(output_channels)):
+        >>>     ch_map_out[output_channels[ndx]]=ndx
+        >>> mc_out = sd.MacCoreSettings(channel_map=ch_map_out)
+        >>>
+        >>> sd.playrec(..., channels=2, extra_settings=(mc_in, mc_out))
+        """
+        # Check inputs:
+        conversion_dict = {
+            'Min':    _lib.paMacCoreConversionQualityMin   ,
+            'Low':    _lib.paMacCoreConversionQualityLow   ,
+            'Medium': _lib.paMacCoreConversionQualityMedium,
+            'High':   _lib.paMacCoreConversionQualityHigh  ,
+            'Max':    _lib.paMacCoreConversionQualityMax   ,
+            }
+        if rate_conv_quality not in conversion_dict:
+            raise ValueError('rate_conv_quality must be one of {!r}'.format(list(conversion_dict.keys())))
+
+        if channel_map is not None:
+            if not all(isinstance(ch, int) for ch in channel_map):
```

Comment:
This check is wrong in several ways.

You are already assuming an iterable by iterating over it, so the check doesn't catch non-iterables.

If you *strictly* want to accept an iterable (and not only, e.g. a sequence), you should only iterate over it not more often than once!
You are iterating once for the type check and later you are passing it on to CFFI, which iterates again.

Checking for a list of `int` is IMHO not necessary, since CFFI does that anyway.
You should check for a single int though, as I mentioned somewhere else before.
You should probably also check for an empty list, since this causes a strange error:

    ||PaMacCore (AUHAL)|| Error on line 1589: err='-50', msg=Unknown Error